tests/providers/openai/test_openai.py

```diff
@@ -0,0 +1,39 @@
+from typing import Tuple
+
+import os
+import pytest
+
+from exchange import Text
+from exchange.message import Message
+from exchange.providers.base import Usage
+from exchange.providers.openai import OpenAiProvider
+from .conftest import OPENAI_MODEL, OPENAI_API_KEY
+
+
+@pytest.mark.vcr()
+def test_openai_completion(monkeypatch):
+    # When running VCR tests the first time, it needs OPENAI_API_KEY to call
+    # the real service. Afterward, it is not needed as VCR mocks the service.
+    if "OPENAI_API_KEY" not in os.environ:
```

Comment:
I guess this will need to be moved to a fixture eventually. Though if we feel there is no practical reason to unset the variable, could consider just code at the module level.

Reply:
oh it is needed because the provider itself is designed to crash if it isn't set. but good point this is better documented as a fixture

Reply:
specifically what's different than default openai is this.. normally we don't need to worry because a real call would fail at http layer with unauthorized. What's causing this issue is we are checking before that https://github.com/square/exchange/blob/main/src/exchange/providers/openai.py#L42

Reply:
PTAL